src/rdb.c

```diff
@@ -2532,9 +2559,16 @@ int rdbLoadRio(rio *rdb, int rdbflags, rdbSaveInfo *rsi) {
         if ((key = rdbGenericLoadStringObject(rdb,RDB_LOAD_SDS,NULL)) == NULL)
             goto eoferr;
         /* Read value */
-        if ((val = rdbLoadObject(type,rdb,key,db->id)) == NULL) {
+        int error;
+        if ((val = rdbLoadObject(type,rdb,key,db->id,&error)) == NULL) {
             sdsfree(key);
-            goto eoferr;
+
+            /* Don't fail when empty key is encountered, we will
+             * silently discard it and continue loading. */
+            if (error == RDB_LOAD_ERR_EMPTY_KEY)
+                continue;
```

Comment:
when we `continue` here, we're skipping this block:
```c
        /* Reset the state that is key-specified and is populated by
         * opcodes before the key, so that we start from scratch again. */
        expiretime = -1;
        lfu_freq = -1;
        lru_idle = -1;
```

maybe a cleaner idea is to add a first `if` to the `if-else` chain below, so that all the other `if`s will not be taken. then similarly to an already expired key, we continue with the normal flow of this loop.

also, let's mention `#8453` in the comment to justify why we do that.

Reply:
Yes, I am building this rdb file.